opencti-platform/opencti-graphql/src/manager/ingestionManager.ts

```diff
@@ -50,50 +49,90 @@ const asArray = (data: unknown) => {
+  'content:encoded': { _: string }
+  category: { _: string } | { _: string }[]
+  pubDate: { _: string }
+  lastBuildDate: { _: string }
+  updated: { _: string }
+}
+
+interface DataItem {
   title: string
   description: string
-  link: string
+  link: string | undefined
   content: string
-  'content:encoded': string
-  category: string | string[]
-  pubDate: string
-  lastBuildDate: string
+  labels: string[]
+  pubDate: Moment
 }
 
-const rssItemConvert = (turndownService: TurndownService, channel: RssElement, item: RssItem) => {
+const rssItemV1Convert = (turndownService: TurndownService, feed: RssElement, entry: RssItem): DataItem => {
+  const { updated } = feed;
+  return {
+    title: turndownService.turndown(entry.title._),
+    description: turndownService.turndown(entry.summary?._ ?? ''),
+    link: isNotEmptyField(entry.link) ? entry.link.href?.trim() : '',
+    content: turndownService.turndown(entry.content?._ ?? ''),
+    labels: [],
+    pubDate: utcDate(entry.updated?._ ?? updated?._),
+  };
+};
+
+const rssItemV2Convert = (turndownService: TurndownService, channel: RssElement, item: RssItem): DataItem => {
   const { pubDate } = channel;
   return {
-    title: turndownService.turndown(item.title),
-    description: turndownService.turndown(item.description),
-    link: item.link,
-    content: turndownService.turndown(item['content:encoded'] ?? item.content ?? ''),
-    labels: R.uniq(asArray(item.category).map((c) => c.trim())),
-    pubDate: utcDate(item.pubDate ?? pubDate),
+    title: turndownService.turndown(item.title._),
+    description: turndownService.turndown(item.description?._ ?? ''),
+    link: isNotEmptyField(item.link) ? (item.link._ ?? '').trim() : '',
+    content: turndownService.turndown(item['content:encoded']?._ ?? item.content?._ ?? ''),
+    labels: R.uniq(asArray(item.category).filter((c) => isNotEmptyField(c)).map((c) => c._.trim())),
+    pubDate: utcDate(item.pubDate?._ ?? pubDate?._),
   };
 };
+
 const rssHttpGetter = (): Getter => {
   const httpClient = getHttpClient({ responseType: 'text' });
   return async (uri: string) => {
     const { data } = await httpClient.get(uri);
     return data;
   };
 };
+
+export const rssDataParser = async (turndownService: TurndownService, data: convertableToString, current_state_date: Date | undefined): Promise<DataItem[]> => {
+  const xmlData = await xmlParse(data, { explicitArray: false, trim: true, explicitCharkey: true, mergeAttrs: true });
+  if (xmlData?.feed) { // Atom V1
+    const entries = asArray(xmlData.feed.entry);
+    return entries.map((entry) => rssItemV1Convert(turndownService, xmlData.feed, entry))
+      .filter((e) => isNotEmptyField(e.title)) // Title is mandatory
+      .filter((e) => isNotEmptyField(e.pubDate) && e.pubDate.isValid()) // A valid date is also mandatory
+      .filter((e) => isEmptyField(current_state_date) || e.pubDate.isAfter(current_state_date))
+      .sort((a, b) => a.pubDate.diff(b.pubDate));
+  }
```

Comment:
We could extract this logic in a function and name it to explicit the comment on the condition

Reply:
Filtering extracted to specific method.